src/main.rs

```diff
@@ -350,8 +350,10 @@ async fn main() -> anyhow::Result<()> {
     }))
     .await
     .context("failed to initialize host")?;
+    let mut stream = signal::unix::signal(signal::unix::SignalKind::terminate())?;
```

Comment:
This should be guarded by `cfg(unix)`, also can we use a more descriptive name? E.g. `terminate`?